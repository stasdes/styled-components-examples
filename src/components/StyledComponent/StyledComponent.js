import styled from 'styled-components'
import styles from './StyledComponent.styles'
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

class StyledComponent extends PureComponent {
  static propTypes = {
    background: PropTypes.string,
    className: PropTypes.string.isRequired,
    children: PropTypes.node
  }

  state = { count: 0 }

  onClick = () => {
    this.setState({count: this.state.count + 1 })
  }

  render () {
    const { className, children } = this.props
    return <div className={className} onClick={this.onClick}>
      <span className='count'>Count {this.state.count}</span> 
      {children}
    </div>
  }
}
export default styled(StyledComponent)`${styles}`
