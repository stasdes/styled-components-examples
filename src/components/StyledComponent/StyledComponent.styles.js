import { css, keyframes } from 'styled-components'
import { media } from '../../utils/mediaQueryUtils'
import StyledSpan2 from '../StyledSpan/StyledSpan2'

const fontSize = '2rem'

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`

export default css`
  /* usage of props */
  background: ${props => props.background};

  /* usage of variables */
  font-size: ${fontSize};

  /* inner style */
  > span.count {
    margin-right: 10px;
    border: 1px solid green;

    /* metaselector  */
    &:hover {
      /* animation example with keyframes */
      animation: 3s ${fadeIn} ease-out;
    }
  }

  /* descendant selector with another styled component */
  > ${StyledSpan2} {
    font-size: ${fontSize};
    color: initial;
  }

  /* Contextual selector */
  .Special & {
    font-weight: bold;
    margin-top: 1rem;
  }

  /* media query hard coded */
  @media (max-width: 990px) {
    background: #db7093;
  }

  /* media query with helper template */
  ${media.tablet`
    background: #00ffff;
  `}

  ${media.phone`
    background: #f1e05a;
  `}
`
