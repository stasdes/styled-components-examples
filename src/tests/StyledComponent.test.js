import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import StyledComponent from '../components/StyledComponent/StyledComponent'

test('StyledComponent should render correctly', () => {
  const tree = renderer.create(<StyledComponent background='#fff' />).toJSON()
  expect(tree).toMatchSnapshot()
})