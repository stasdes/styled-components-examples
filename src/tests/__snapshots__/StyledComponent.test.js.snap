// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledComponent should render correctly 1`] = `
.c0 {
  background: #fff;
  font-size: 2rem;
}

.c0 > span.count {
  margin-right: 10px;
  border: 1px solid green;
}

.c0 > span.count:hover {
  -webkit-animation: 3s bcCCNc ease-out;
  animation: 3s bcCCNc ease-out;
}

.c0 > .sc-bdVaJa {
  font-size: 2rem;
  color: initial;
}

.Special .c0 {
  font-weight: bold;
  margin-top: 1rem;
}

@media (max-width:990px) {
  .c0 {
    background: #db7093;
  }
}

@media (max-width:768px) {
  .c0 {
    background: #00ffff;
  }
}

@media (max-width:500px) {
  .c0 {
    background: #f1e05a;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <span
    className="count"
  >
    Count 
    0
  </span>
</div>
`;
